lib/cretonne/src/ir/dfg.rs

```diff
@@ -218,6 +218,33 @@ impl DataFlowGraph {
             panic!("Cannot change direct value {} into an alias", dest);
         }
     }
+
+    /// Create a new value alias.
+    pub fn make_value_alias(&mut self, dest: Value, src: Value) -> Value {
```

Comment:
This function should not take a `dest` argument. That is the newly created value it will be returning.

This also means that it condenses down to calling `value_type` and `make_value`.

I think it would be a good idea to mention in the function's doc comment that it only is used by the parser. There shouldn't be another reason for creating aliases during compilation. (Other than `change_to_alias`).